examples/passport-login/src/__tests__/acceptance/passport-login.acceptance.ts

```diff
@@ -0,0 +1,406 @@
+          .auth('test@example.com', 'incorrect-password', {type: 'basic'})
+          .expect(401);
+      });
+    });
+
+    /**
+     *********************************************************
+     *  Scenario 2: Link an external profile with a local user
+     *********************************************************
+     *    Test case 1: login via a social app profile having same email id as local user
+     *    Test case 2: check if external profile is linked to local user
+     *    Test case 3: logout
+     */
+    context('Scenario 2: Link an external profile with a local user', () => {
+      let oauthProviderUrl: string;
+      let providerLoginUrl: string;
+      let loginPageParams: string;
+      let callbackToLbApp: string;
+
+      it('call is redirected to third party authorization url', async () => {
+        const response = await client
+          .get('/api/auth/thirdparty/oauth2')
+          .expect(303);
+        oauthProviderUrl = response.get('Location');
+        expect(url.parse(oauthProviderUrl).pathname).to.equal('/oauth/dialog');
+      });
+
+      it('call to authorization url is redirected to oauth providers login page', async () => {
+        const response = await supertest('').get(oauthProviderUrl).expect(302);
+        providerLoginUrl = response.get('Location');
+        loginPageParams = url.parse(providerLoginUrl).query ?? '';
+        expect(url.parse(response.get('Location')).pathname).to.equal('/login');
+      });
+
+      /**
+       * Sign Up via a social app with the following profile
+       *   username: testuser
+       *   email: test@example.com
+       *
+       * Email-id MATCHES local account
+       */
+      it('login page redirects to authorization app callback endpoint', async () => {
+        const loginPageHiddenParams = qs.parse(loginPageParams);
+        const params = {
+          username: 'testuser',
+          password: 'xyz',
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          client_id: loginPageHiddenParams.client_id,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          redirect_uri: loginPageHiddenParams.redirect_uri,
+          scope: loginPageHiddenParams.scope,
+        };
+        // On successful login, the authorizing app redirects to the callback url
+        // HTTP status code 302 is returned to the browser
+        const response = await supertest('')
+          .post('http://localhost:9000/login_submit')
+          .send(qs.stringify(params))
+          .expect(302);
+        callbackToLbApp = response.get('Location');
+        expect(url.parse(callbackToLbApp).pathname).to.equal(
+          '/api/auth/thirdparty/oauth2/callback',
+        );
+      });
+
+      it('callback url contains access code', async () => {
+        expect(url.parse(callbackToLbApp).query).to.containEql('code');
+      });
+
+      it('access code can be exchanged for token', async () => {
```

Comment:
hmm how does the code and token exchange get tested in it?

Reply:
authorization code exchange test is also done in acceptance tests of extension-passport